### PR TITLE
Improve closer implementation and docstrings

### PR DESCRIPTION
**Pull request by gdb**
@jietang mind reviewing?

### CODE REVIEW

File: gym/core.py [hunk 1]
```diff
@@ -31,13 +33,22 @@ class Env(object):
 
         action_space: The Space object corresponding to valid actions
         observation_space: The Space object corresponding to valid observations
+        reward_range: A tuple corresponding to the min and max possible rewards
```
Comment (jietang) [inline gym/core.py]: It doesn't look like we actually use this anywhere - should we get rid of it?

File: gym/core.py [hunk 2]
```diff
@@ -58,12 +69,6 @@ def _render(self, mode='human', close=False):
             return
         raise NotImplementedError
 
-    # Will be automatically set when creating an environment via
```
Comment (jietang) [inline gym/core.py]: maybe worth keeping this comment in __new__ so it's easier to trace the magic that sets spec

File: gym/core.py (hunk 1 above)
Comment (gdb) [inline gym/core.py]: This is for use by people's algorithms.

File: gym/tests/test_core.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+class ArgumentEnv(core.Env):
+    calls = 0
+
+    def __init__(self, arg):
+        self.calls += 1
+        self.arg = arg
+
+def test_env_instantiation():
+    # This looks like a pretty trivial, but given our usage of
```
Comment (jietang) [inline gym/tests/test_core.py]: +1

File: gym/utils/closer.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+class Closer(object):
+    """A registry that ensures your objects get closed, whether manually,
+    upon garbage collection, or upon exit. To work properly, your
+    objects need to cooperate and do something like the following:
+
+    ```
+    closer = Closer()
+    class Example(object):
+        def __init__(self):
+        self._id = closer.register(self)
```
Comment (jietang) [inline gym/utils/closer.py]: indentation looks off

File: gym/core.py (hunk 2 above)
Comment (gdb) [inline gym/core.py]: Good call.

**Comment by jietang** [comment]
lgtm

**Pull request merged**